### ISSUE TITLE
On headed Linux: `Driver(headless=True, undetectable=True)` isn't using headless mode

### ISSUE THREAD

**Issue by jdholtz** [seleniumbase/SeleniumBase#2176]
Hello. I am currently using SeleniumBase 4.20.0 with Python 3.11.5. When I execute the following code, the browser is still headed:
```py
from seleniumbase import Driver
driver = Driver(headless=True, driver_version="browser", undetectable=True)
```

I am currently using Chromium 118.0.5993.54 on Arch Linux, but I have noticed this issue with Chromium v116 and v117 ever since this driver version option was added.

Since `headless=True` is specified, I expect the browser to still be headless, no matter what driver version is specified.

**Comment by mdmintz** [comment]
Your question title has `driver_version="keep"`, but your question description has `driver_version="browser"`, which are two different options. You also mentioned that you're using raw `Chromium`, but you need to specifically use `Chrome` when using UC Mode. You probably hit an error for that, and when that happens, none of the settings provided get used. Make sure that standard Chrome is installed or your system.

<img width="700" alt="Screenshot 2023-10-10 at 11 13 46 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/8b6e3762-0235-4e70-b7d1-e27bddd0818e">

--------

I ran the exact code from your description, and it ran correctly in UC headless mode on my local computer.

Remember, UC is short for Undetected Chrome, **not** Undetected Chromium!

**Comment by jdholtz** [comment]
> Your question title has driver_version="keep", but your question description has driver_version="browser", which are two different options

That is my bad, I will edit the title. I meant `driver_version="browser"`. 

> You also mentioned that you're using raw Chromium, but you need to specifically use Chrome when using UC Mode.

Everything has worked perfectly fine with Chromium, but I installed Chrome and removed Chromium just to test and am still running into the same issue. I attached a video to see the issue more clearly.

> Also, your Chrome version must be a minimum of 115.X.X.X in order for driver_version="browser" to work

Here is the output of `google-chrome-stable --version`: Google Chrome 118.0.5993.70 so it should be good.

https://github.com/seleniumbase/SeleniumBase/assets/89617856/05bf0cb7-a447-4c77-b406-bd88f0a933de

**Comment by jdholtz** [comment]
I did some more debugging and figured out what the issue is. On line 2833 of [seleniumbase/core/browser_launcher.py](https://github.com/seleniumbase/SeleniumBase/blob/cd66f4a0a834767ea3e9b842ead5658dd838b714/seleniumbase/core/browser_launcher.py#L2833), use_version is the browser version (118.0.5993.70) which causes an exception to be thrown when converting it to an int and therefore the headless argument is never added.

Here is a diff of what fixes it for me. Let me know if you want me to make a PR for this.
```diff
                 try:
-                    if int(use_version) >= 109:
+                    major_version = use_version.split(".")[0]
+                    if int(major_version) >= 109:
                         chrome_options.add_argument("--headless=new")
                     elif (
-                        int(use_version) >= 96
-                        and int(use_version) <= 108
+                        int(major_version) >= 96
+                        and int(major_version) <= 108
                     ):
```

**Comment by mdmintz** [comment]
Try using `headless2` instead of `headless` in your call. Regular headless mode should never have been reached that point because UC Mode doesn't support it:

```python
from seleniumbase import Driver
driver = Driver(headless2=True, undetectable=True)
```

**Comment by mdmintz** [comment]
Oh, I think I see the issue. Hold on, working on a fix...

**Comment by mdmintz** [comment]
I changed the description:

"On headed Linux: `Driver(headless=True, undetectable=True)` isn't using headless mode"

If the Linux machine had been headless, Xvfb would've activated. This is a special case for headed Linux.

**Comment by jdholtz** [comment]
> Try using headless2 instead of headless in your call. Regular headless mode should never have been reached at that point because UC Mode doesn't support it:

Using headless2 results in me being detected whereas headless doesn't. 

> "On headed Linux: Driver(headless=True, undetectable=True) isn't using headless mode"

This line actually correctly uses headless mode for me. It is only when the `driver_version` is specified as the entire version (XXX.X.XXXX.XX or "browser") that it doesn't use headless mode (my diff fixes that).

**Comment by mdmintz** [comment]
I'll figure it out in the morning. It's past 2AM here.

**Comment by mdmintz** [comment]
I have the fix, and some other improvements to be part of the next release.

**Comment by mdmintz** [comment]
This was resolved in `4.20.1` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.20.1

**Comment by jdholtz** [comment]
Thanks a lot for the quick fix!
